com.unity.ml-agents/Runtime/DecisionRequester.cs

```diff
@@ -57,21 +65,58 @@ void OnDestroy()
             }
         }
 
+        /// <summary>
+        /// Information about Academy step used to make decisions about whether to request a decision.
+        /// </summary>
+        public struct DecisionRequestContext
+        {
+            /// <summary>
+            /// The current step count of the Academy, equivalent to Academy.StepCount.
+            /// </summary>
+            public int AcademyStepCount;
+        }
+
         /// <summary>
         /// Method that hooks into the Academy in order inform the Agent on whether or not it should request a
         /// decision, and whether or not it should take actions between decisions.
         /// </summary>
         /// <param name="academyStepCount">The current step count of the academy.</param>
         void MakeRequests(int academyStepCount)
         {
-            if (academyStepCount % DecisionPeriod == 0)
+            var context = new DecisionRequestContext
+            {
+                AcademyStepCount = academyStepCount
+            };
+
+            if (ShouldRequestDecision(context))
             {
                 m_Agent?.RequestDecision();
             }
-            if (TakeActionsBetweenDecisions)
+
+            if (ShouldRequestAction(context))
             {
                 m_Agent?.RequestAction();
             }
         }
+
+        /// <summary>
+        /// Whether Agent.RequestDecision should be called on this update step.
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        protected virtual bool ShouldRequestDecision(DecisionRequestContext context)
```

Comment:
The `DecisionRequestContext` is maybe a little over-engineered, but it'll let us extend in the future if needed without changing the signature.